esphome/components/ld2410s/ld2410s.h

```diff
@@ -0,0 +1,233 @@
+
+  std::string energy_values_str_ = "";
+  std::vector<LD2410SListener *> listeners_{};
+  CmdT commands_[CMD_EXEC_BUFFER_SIZE];
+  TriggersT triggers_;
+
+  void init_();
+  void loop_send_command_();
+  void cmd_buffer_insert_(CmdFrameT *cmd_frame);
+  void cmd_buffer_finished_();
+  void cmd_buffer_inc_(uint8_t &index);
+
+#ifdef USE_NUMBER
+  number::Number *max_distance_number_{nullptr};
+  number::Number *min_distance_number_{nullptr};
+  number::Number *no_delay_number_{nullptr};
+  number::Number *status_reporting_freq_number_{nullptr};
+  number::Number *distance_reporting_freq_number_{nullptr};
+
+  number::Number *trigger_threshold_number_{nullptr};
+  number::Number *trigger_hold_number_{nullptr};
+  number::Number *trigger_snr_number_{nullptr};
+  number::Number *trigger_selected_gate_number_{nullptr};
+#endif
+
+#ifdef USE_BUTTON
+  button::Button *read_all_button_{nullptr};
+  button::Button *write_all_button_{nullptr};
+  button::Button *calibration_button_{nullptr};
+  button::Button *factory_reset_button_{nullptr};
+  button::Button *toggle_minimal_output_button_{nullptr};
+#endif
+
+#ifdef USE_SELECT
+  select::Select *response_speed_select_{nullptr};
+#endif
+
+  void receive_();
+  PackageType get_frame_type_(uint8_t *buffer, size_t pos);
+  size_t get_frame_start_(uint8_t *buffer, size_t end_pos, PackageType type);
+  size_t get_payload_size_(uint8_t *buffer, size_t end_pos, PackageType type, size_t start_pos);
+
+  void process_short_data_frame_(uint8_t *data);
+  void process_data_frame_(uint8_t *data, size_t data_size);
+  void process_cmd_frame_(uint8_t *buffer, size_t len);
+
+  CmdAckT parse_cms_frame_(uint8_t *buffer, size_t length);
+
+  void schedule_cmd_(const char *msg, uint16_t command, uint16_t sub_command = NO_SUB_CMD);
+  void schedule_cmd_frame_(uint16_t command, uint16_t sub_command = NO_SUB_CMD);
+  void cmd_frame_append_data_(CmdFrameT *cmd_frame, const uint8_t *append_data, size_t append_data_size);
+  void cmd_frame_append_data_(CmdFrameT *cmd_frame, const uint16_t *append_data, size_t append_data_size);
+  void cmd_frame_append_data_(CmdFrameT *cmd_frame, const uint32_t *append_data, size_t append_data_size);
+
+  void send_command_(CmdFrameT *cmd_frame);
+
+  void process_ack_config_read_(uint8_t *data);
+  void process_ack_fw_read_(const uint8_t *data);
+  void process_ack_trigger_threshold_read_(uint8_t *data);
+  void process_ack_trigger_hold_read_(uint8_t *data);
+  void process_ack_trigger_snr_read_(uint8_t *data);
+  void process_data_energy_values_read_(uint8_t *data);
+
+  void update_ts_thresholds_();
+  void update_ts_holds_();
+  void update_ts_snrs_();
+  void update_ts_energy_values_();
+
+  std::string format_int_(uint32_t *in, uint8_t len, uint8_t min_w);
+  void four_byte_to_int_array_(uint8_t *in, uint32_t *out, uint8_t out_len);
+  void hex_diag_(const char *msg, const uint8_t *data, size_t length);
+  int read_int_(const uint8_t *buffer, size_t pos, size_t len);
```

Comment:
As these don't appear to operate on anything specific in the class, please make them static.
```suggestion
  static std::string format_int_(uint32_t *in, uint8_t len, uint8_t min_w);
  static void four_byte_to_int_array_(uint8_t *in, uint32_t *out, uint8_t out_len);
  static void hex_diag_(const char *msg, const uint8_t *data, size_t length);
  static int read_int_(const uint8_t *buffer, size_t pos, size_t len);
```